### ISSUE TITLE
Illegal Instruction - `python3 -m markov.rollout_worker`

### ISSUE THREAD

**Issue by whatcanising-admin** [aws-deepracer-community/deepracer-core#35]
In trying to run the deepracer_robomaker:console docker image the whole process crashes inside docker. 

![image](https://user-images.githubusercontent.com/49555088/61603361-91000600-ac80-11e9-8f51-02991153c52a.png)

Tried to debug by executing bash against the docker image. 

_cd /app/robomaker-deepracer/simulation_ws/install/deepracer_simulation/lib/deepracer_simulation
python3 -m markov.rollout_worker
Illegal instruction (core dumped)_

Host is a Ubuntu 18.04.2 LTS on a VMWare ESXi stack. 
16 Cores and 64GB RAM 

Docker 18.06.1-ce
Python 3.6.8

**Comment by crr0004** [comment]
Are you running that command directly? Normally ROS runs that command once everything is setup. I suppose you can run the files directly but it won't get very far. One the big issues of this setup is debugging. It's been a nightmare as it's near impossible to get a debug hook into it.

**Comment by whatcanising-admin** [comment]
Not running directly, running as per the instructions for the docker container.

**Comment by zibjdp** [comment]
I picked up the same issue.
Summary:
Tensorflow requires AVX instruction set on the CPU that is not available on older CPUs and will break Tensorflow.

Details:
Robotmaker docker container not starting correctly:

Docker logs extract:
```
/app/robomaker-deepracer/simulation_ws/install/deepracer_simulation/lib/deepracer_simulation/run_rollout_rl_agent.sh: line 8:  1527 Illegal instruction     (core dumped) python3 -m markov.rollout_worker
================================================================================REQUIRED process [agent-9] has died!
process has died [pid 1402, exit code 132, cmd /app/robomaker-deepracer/simulation_ws/install/deepracer_simulation/lib/deepracer_simulation/run_rollout_rl_agent.sh __name:=agent __log:=/root/.ros/log/a7aa9c0a-c367-11e9-94b9-0242ac120004/agent-9.log].
log file: /root/.ros/log/a7aa9c0a-c367-11e9-94b9-0242ac120004/agent-9*.log
Initiating shutdown!
================================================================================
```
using 'docker exec -ti robomaker bash' to run python3 -v -m markov.rollout_worker in the container shows that the 'Illegal instruction' happens when tensorflow is imported.

Digging around on the internet revealed that tensorflow from v1.6.0 is prebuilt with AVX instructions available in modern CPUs.
https://stackoverflow.com/questions/49122044/illegal-instruction-when-import-tensorflow-in-python
https://github.com/tensorflow/tensorflow/releases/tag/v1.6.0

checking my CPU features with 'cat /proc/cpuinfo' showed that my Intel i7 920 (released 2008) does not have the AVX instructions.
https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#CPUs_with_AVX
If your CPU model was released before 2011, then it will not support AVX and will fail with the version of Tensorflow(1.11.0) in the robomaker container.

It looks like there is a way to compile newer versions of tensorflow without AVX support. But I have not tried that option yet.

Hope this helps other users getting the same Illegal instruction error
